pkg/devicestate/state.go

```
@@ -214,6 +214,67 @@ func (s *Manager) applyConfigOnDevice(ctx context.Context, ifNameIndex *int, cla
 		})
 	}
 
+	// If device is RDMA capable, add RDMA character devices
+	if rdmaCapableAttr, ok := deviceInfo.Attributes[consts.AttributeRDMACapable]; ok && rdmaCapableAttr.BoolValue != nil && *rdmaCapableAttr.BoolValue {
+		rdmaDevices, err := host.GetHelpers().GetRDMADeviceForPCI(pciAddress)
+		if err != nil {
+			logger.Error(err, "Failed to get RDMA devices for PCI address",
```

I think we should return an error here and not let the pod start without the RDMA devices.

if the VF is reported as RDMA this means we manage to run the `GetRDMADeviceForPCI` in the discovery.